lib/matplotlib/tests/test_backends_interactive.py

```diff
@@ -216,16 +214,19 @@ def _test_thread_impl():
 
 _thread_safe_backends = _get_testable_interactive_backends()
 # Known unsafe backends. Remove the xfails if they start to pass!
-if "wx" in _thread_safe_backends:
-    _thread_safe_backends.remove("wx")
-    _thread_safe_backends.append(
-        pytest.param("wx", marks=pytest.mark.xfail(
-            raises=subprocess.CalledProcessError)))
-if "macosx" in _thread_safe_backends:
-    _thread_safe_backends.remove("macosx")
-    _thread_safe_backends.append(
-        pytest.param("macosx", marks=pytest.mark.xfail(
-            raises=subprocess.TimeoutExpired, strict=True)))
+for param in _thread_safe_backends:
+    backend = param.values[0]
+    if "cairo" in backend:
+        # Cairo backends save a cairo_t on the graphics context, and sharing
+        # these is not threadsafe.
+        param.marks.append(
```

Comment:
I don't understand what this is doing.

Reply:
I was reading this as `pytest.marks` not `param.marks` for a good 5 minutes :sheep: